### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by Rombur**
Rename `BBox` to `Box` and add ETI to details. Should we rename all the files in `details` as `DTK_DetailsXXX`?

### CODE REVIEW

File: packages/Search/cmake/ExplicitInstantiationSupport.cmake [hunk 1]
```diff
@@ -94,4 +94,7 @@ TRIBITS_ETI_GENERATE_TYPEDEF_MACRO(DTK_ETI_TYPEDEFS "DTK_ETI_MANGLING_TYPEDEFS"
 # ${PACKAGE_NAME}_ETIHelperMacros.h.in
 CONFIGURE_FILE(
   ${DataTransferKit_SOURCE_DIR}/packages/Search/cmake/${PACKAGE_NAME}_ETIHelperMacros.h.in
+  ${DataTransferKit_BINARY_DIR}/packages/Search/src/details/${PACKAGE_NAME}_ETIHelperMacros.h)
+CONFIGURE_FILE(
+  ${DataTransferKit_SOURCE_DIR}/packages/Search/cmake/${PACKAGE_NAME}_ETIHelperMacros.h.in
```
Comment (aprokop) [inline packages/Search/cmake/ExplicitInstantiationSupport.cmake]: This line is duplicate of 2 above.

File: packages/Search/cmake/ExplicitInstantiationSupport.cmake [hunk 2]
```diff
@@ -94,4 +94,7 @@ TRIBITS_ETI_GENERATE_TYPEDEF_MACRO(DTK_ETI_TYPEDEFS "DTK_ETI_MANGLING_TYPEDEFS"
 # ${PACKAGE_NAME}_ETIHelperMacros.h.in
 CONFIGURE_FILE(
   ${DataTransferKit_SOURCE_DIR}/packages/Search/cmake/${PACKAGE_NAME}_ETIHelperMacros.h.in
+  ${DataTransferKit_BINARY_DIR}/packages/Search/src/details/${PACKAGE_NAME}_ETIHelperMacros.h)
```
Comment (aprokop) [inline packages/Search/cmake/ExplicitInstantiationSupport.cmake]: Why do we also need `details/...ETIHElperMacros.h`? This package has already generated one for us using the standard path.

File: packages/Search/src/details/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+APPEND_GLOB(HEADERS ${DIR}/*.h)
+APPEND_GLOB(HEADERS ${DIR}/*.hpp)
+APPEND_GLOB(SOURCES ${DIR}/*.cpp)
+TRILINOS_CREATE_CLIENT_TEMPLATE_HEADERS(${DIR})
+
+# Must glob the binary dir last to get all of the auto-generated headers
+SET_AND_INC_DIRS(DIR ${CMAKE_CURRENT_BINARY_DIR})
+APPEND_GLOB(HEADERS ${DIR}/*.hpp)
+APPEND_SET(HEADERS ${DIR}/${PACKAGE_NAME}_config.h)
+APPEND_SET(HEADERS ${DIR}/${PACKAGE_NAME}_ETIHelperMacros.h)
```
Comment (aprokop) [inline packages/Search/src/details/CMakeLists.txt]: Can we replace this by `${CMAKE_CURRENT_BINARY_DIR}/../${PACKAGE_NAME}_ETIHelperMacros.h`?

File: packages/Search/src/details/DTK_ETI_NT.tmpl [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+// CMake takes each expression in the original .tmpl file enclosed by
+// "at" symbols ("a" with a circle around it), and replaces it in the
+// generated .cpp file with a string defined by Tpetra's CMake logic
+// (see CMakeLists.txt in this directory).  Thus, the original .tmpl
+// file is NOT syntactically correct C++, but the .cpp file generated
+// by running CMake on it IS a syntactically correct C++ file.
+
+#include "DataTransferKitSearch_config.h"
+
+#if defined(HAVE_DATATRANSFERKIT_EXPLICIT_INSTANTIATION)
```
Comment (aprokop) [inline packages/Search/src/details/DTK_ETI_NT.tmpl]: Is there no other way than to copy the `*.tmpl` files?

**Comment by Rombur** [comment]
A lot of the redundancy that we have in the files is actually required by Tribits. I have removed everything that is not necessary in the cmake.

**Comment by aprokop** [comment]
Should `DTK_ETI_SC_LO_GO_NT.tmpl` and the corresponding CMake code be removed? We don't template anything on `LO` and `GO` anyway.

**Comment by sslattery** [comment]
yes I think we should remove that file

**Comment by Rombur** [comment]
OK done

**Comment by aprokop** [comment]
Related to #217.

**Pull request merged**